File: api/src/main/java/com/delivery/common/exception/ApiException.java
package com.delivery.common.exception;

import com.delivery.common.status.StatusCode;
import lombok.Getter;

@Getter
public class ApiException extends RuntimeException {
    private final StatusCode errorStatusCode;

    private final String message;

    public ApiException(StatusCode errorStatusCode) {
        super(errorStatusCode.getMessage());
        this.errorStatusCode = errorStatusCode;
        this.message = errorStatusCode.getMessage();
    }

    public ApiException(StatusCode errorStatusCode, String message) {
        super(errorStatusCode.getMessage());
        this.errorStatusCode = errorStatusCode;
        this.message = message;
    }

    public ApiException(StatusCode errorStatusCode, Throwable t) {
        super(t);
        this.errorStatusCode = errorStatusCode;
        this.message = errorStatusCode.getMessage();
    }
}

File: common/src/main/kotlin/com/delivery/common/exception/ApiException.kt
package com.delivery.common.exception

import com.delivery.common.status.ErrorStatusCode
import com.delivery.common.status.StatusCode

class ApiException : RuntimeException {

    val errorStatusCode: StatusCode
    val errorMessage: String

    companion object {
        @JvmStatic
        fun nullError(): ApiException {
            return ApiException(ErrorStatusCode.NULL_POINT)
        }
    }

    constructor(errorStatusCode: StatusCode) : super(errorStatusCode.getMessage()) {
        this.errorStatusCode = errorStatusCode
        errorMessage = errorStatusCode.getMessage()
    }

    constructor(
        errorStatusCode: StatusCode,
        message: String,
    ) : super(errorStatusCode.getMessage()) {
        this.errorStatusCode = errorStatusCode
        this.errorMessage = message
    }

    constructor(
        errorStatusCode: StatusCode,
        t: Throwable,
    ) : super(t) {
        this.errorStatusCode = errorStatusCode
        this.errorMessage = errorStatusCode.getMessage()
    }

}